api/agent.go

```diff
@@ -602,6 +601,7 @@ func (a *Agent) AgentHealthServiceByIDOpts(serviceID string, q *QueryOptions) (s
 	r.setQueryOptions(q)
 	r.params.Add("format", "json")
 	r.header.Set("Accept", "application/json")
+	// not a lot of value in wrapping the doRequest call in a requireHttpCodes call
```

Comment:
Good point! There's two ways I understand this comment:

1. Should we return a `StatusError` from the `require` calls and handle them inside the body of the func?

2. Should we return a `StatusError` our of this func?

Reply:
Let's chat about 1) first.

From a cursory look, I was worried that we'd change the return format.

So I was thinking that some of callees of these two API paths (ie `/v1/agent/health/service/id/%v`) were error handling for the old err messages. 

In order to keep the same format, we'd have to wrap it the same so it'd end up looking something like

```
return HealthCritical, out, fmt.Errorf("Unexpected Error Code %v for %s", err.StatusCode, path)
```
--
That's why I also said that there isn't a lot of value to wrapping the call since there is sufficient "behavior" handling and a different error message returned.

Reply:
For 2), I think we could get away with sending out an `*api.StatusError`. It's just a matter of whether it makes sense to expose that. I've not looked too hard into what changes would be required to do so if any at all.

What do you think ?

Reply:
We should consistently return `api.StatusError`, especially in cases where error strings look similar (otherwise it might mislead clients to believe they're dealing with the custom type)

Reply:
> I was mostly asking about 2), since the error string looks similar to the one from StatusError

> We should consistently return api.StatusError, especially in cases where error strings look similar (otherwise it might mislead clients to believe they're dealing with the custom type)

+1; should be something that I can address per our chat below in a separate PR

merging these changes for now.